lib: unable to link to Zephyr's POSIX API when using a 3rd-party C library

**Describe the bug**
It is currently not possible to link to Zephyr's POSIX API when using a 3rd-party C library.

Apparently, `CONFIG_EXTERNAL_LIBC` should be substituted with `CONFIG_NATIVE_LIBC` below. That way, `lib/posix` does not need to be excluded.

https://github.com/zephyrproject-rtos/zephyr/blob/87d056bd31b14b680e6f011134c7ce433c4462f6/lib/CMakeLists.txt#L7

Please also mention any information which could help others to understand
the problem you're facing:
 - **What target platform are you using?** Any / all
 - **What have you tried to diagnose or workaround this issue?** Encountered errors, gotten feedback from maintainers, and created a PR using the suggested fix.
 - Is this a regression? No

**To Reproduce**
Steps to reproduce the behavior:
1. apply the patch below
2. `west build -p auto -b qemu_riscv64 samples/hello_world`
3. See error

**Expected behavior**
I would expect to be able to include Zephyr's POSIX API. The error goes away with the linked PR.

**Impact**
It's a bit of a showstopper because I would like to use Zephyr's POSIX API.

**Logs and console output**
```shell
/samples/hello_world/src/main.c:17: undefined reference to `timespec_to_timeoutms'
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used

**Additional context**
```diff
diff --git a/samples/hello_world/CMakeLists.txt b/samples/hello_world/CMakeLists.txt
index ecb7d24bb8..d1423df02b 100644
--- a/samples/hello_world/CMakeLists.txt
+++ b/samples/hello_world/CMakeLists.txt
@@ -5,4 +5,4 @@ cmake_minimum_required(VERSION 3.20.0)
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project(hello_world)
 
-target_sources(app PRIVATE src/main.c)
+target_sources(app PRIVATE src/main.c src/libc.c)
diff --git a/samples/hello_world/prj.conf b/samples/hello_world/prj.conf
index b2a4ba5910..759313e918 100644
--- a/samples/hello_world/prj.conf
+++ b/samples/hello_world/prj.conf
@@ -1 +1,2 @@
-# nothing here
+CONFIG_EXTERNAL_LIBC=y
+CONFIG_POSIX_CLOCK=y
diff --git a/samples/hello_world/src/main.c b/samples/hello_world/src/main.c
index 2758d75d3f..5ed15614b0 100644
--- a/samples/hello_world/src/main.c
+++ b/samples/hello_world/src/main.c
@@ -4,10 +4,16 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <errno.h>
+#include <stdint.h>
 #include <stdio.h>
+#include <time.h>
+
+extern int64_t timespec_to_timeoutms(const struct timespec *abstime);
 
 int main(void)
 {
-	printf("Hello World! %s\n", CONFIG_BOARD);
-	return 0;
+	errno = ECHILD;
+	perror("Hi!");
+	return timespec_to_timeoutms(NULL);
 }

```

--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -4,10 +4,10 @@
 add_compile_options($<TARGET_PROPERTY:compiler,warning_shadow_variables>)
 
 add_subdirectory(crc)
-if(NOT CONFIG_EXTERNAL_LIBC)
+if(NOT CONFIG_NATIVE_LIBC)
 add_subdirectory(libc)
-add_subdirectory(posix)
 endif()
+add_subdirectory(posix)
 add_subdirectory_ifdef(CONFIG_CPP                  cpp)
 add_subdirectory(hash)
 add_subdirectory(heap)

